examples/suit_update/tests/01-run.py

```diff
@@ -0,0 +1,128 @@
+# directory for more details.
+
+import os
+import subprocess
+import sys
+import tempfile
+
+from testrunner import run
+
+# Default test over loopback interface
+COAP_HOST = "[fd00:dead:beef::1]"
+
+UPDATE_TIMEOUT = 120
+MANIFEST_TIMEOUT = 15
+
+USE_ETHOS = int(os.getenv("USE_ETHOS", "1"))
+TAP = os.getenv("TAP", "riot0")
+TMPDIR = tempfile.TemporaryDirectory()
+
+
+def start_aiocoap_fileserver():
+    aiocoap_process = subprocess.Popen(
+        "exec aiocoap-fileserver %s" % TMPDIR.name, shell=True
+    )
+
+    return aiocoap_process
+
+
+def cleanup(aiocoap_process):
+    aiocoap_process.kill()
+    TMPDIR.cleanup()
+
+
+def notify(coap_server, client_url, version=None):
+    cmd = [
+        "make",
+        "suit/notify",
+        "SUIT_COAP_SERVER={}".format(coap_server),
+        "SUIT_CLIENT={}".format(client_url),
+    ]
+    if version is not None:
+        cmd.append("SUIT_NOTIFY_VERSION={}".format(version))
+    assert not subprocess.call(cmd)
+
+
+def publish(server_dir, server_url, app_ver, latest_name=None):
+    cmd = [
+        "make",
+        "suit/publish",
+        "SUIT_COAP_FSROOT={}".format(server_dir),
+        "SUIT_COAP_SERVER={}".format(server_url),
+        "APP_VER={}".format(app_ver),
+        "RIOTBOOT_SKIP_COMPILE=1",
+    ]
+    if latest_name is not None:
+        cmd.append("SUIT_MANIFEST_SIGNED_LATEST={}".format(latest_name))
+
+    assert not subprocess.call(cmd)
+
+
+def testfunc(child):
+    """For one board test if specified application is updatable"""
+
+    # Initial Setup and wait for address configuration
+    child.expect_exact("main(): This is RIOT!")
+
+    # get version of currently running image
+    # "Image Version: 0x00000000"
+    child.expect(r"Image Version: (?P<app_ver>0x[0-9a-fA-F:]+)")
+    current_app_ver = int(child.match.group("app_ver"), 16)
+    print("CURRENT APP_VER={}".format(current_app_ver))
+
+    if USE_ETHOS == 0:
+        # Get device global address
+        child.expect(
+            r"inet6 addr: (?P<gladdr>[0-9a-fA-F:]+:[A-Fa-f:0-9]+)"
+            "  scope: global  VAL"
+        )
+        client = "[{}]".format(child.match.group("gladdr").lower())
+    else:
+        # Get device local address
+        client = "[fe80::2%{}]".format(TAP)
+
+    for version in [current_app_ver + 1, current_app_ver + 2]:
+        # Wait for suit_coap thread to start
+        child.expect_exact("suit_coap: started.")
+        # Trigger update process, verify it validates manifest correctly
+        publish(TMPDIR.name, COAP_HOST, version)
+        notify(COAP_HOST, client, version)
+        child.expect_exact("suit_coap: trigger received")
+        child.expect_exact("suit: verifying manifest signature...")
+        child.expect(
+            r"riotboot_flashwrite: initializing update to target slot (\d+)",
+            timeout=MANIFEST_TIMEOUT,
+        )
+        target_slot = int(child.match.group(1))
+        # Wait for update to complete
+        child.expect_exact(
```

Comment:
I was thinking of changing the way it waits for the update to complete to this:

```
def wait_for_update(child):
    try:
        while True:
            child.expect(r"riotboot_flashwrite: processing bytes (\d+)-(\d+)",
                         timeout=TIMEOUT)
    except pexpect.TIMEOUT:
        child.expect_exact(
            "riotboot_flashwrite: riotboot flashing completed successfully",
            timeout=TIMEOUT)
        print(child.after)
```

That way it doesnt depend on a very long timeout that can vary according to the board/cpu.

Reply:
It works for me, any reason why it wouldn't in the CI or why this is a bad idea? @kaspar030

Reply:
This way it'll always timeout at least once (after the update is complete). Isn't it possible to wait for two strings with pexpect?

Reply:
See here: https://pexpect.readthedocs.io/en/stable/api/pexpect.html#pexpect.spawn.expect

You can just pass a list.

How about:

```
while True:
            child.expect([r"riotboot_flashwrite: processing bytes (\d+)-(\d+)", "riotboot_flashwrite: riotboot flashing completed successfully"])
                         timeout=TIMEOUT)
```

Reply:
well, need to check the result:

```
while True:
            index = child.expect([r"riotboot_flashwrite: processing bytes (\d+)-(\d+)",
                                  "riotboot_flashwrite: riotboot flashing completed successfully"])
                         timeout=TIMEOUT)
             if index == 1:
                  break
```

Reply:
I did something like that:

```
def wait_for_update(child):
        return child.expect([r"riotboot_flashwrite: processing bytes (\d+)-(\d+)",
                             "riotboot_flashwrite: riotboot flashing "
                             "completed successfully"],
                             timeout=UPDATING_TIMEOUT)

# Wait for update to complete
while wait_for_update(child) == 0:
     pass
```

Reply:
You ok with the change? Tested and ready to push!

Reply:
yes, nice!